File: ServerFrontEnd/ServerFrontEnd/Operations/Interfaces/RepositoryOperation.cs
namespace ServerFrontEnd.Operations.Interfaces;

public class RepositoryOperation
{
    public String Repository { get; set; }
    public String State { get; set; }
    public String Type { get; set; }

    private String result;
    public String Result
    {
        get
        {
            result = ResultToJson();
            return result;
        }
        set
        {
            result = value;
            JsonToResult(value);
        }
    }

    public RepositoryOperation(string repository, string state, string type)
    {
        Repository = repository;
        State = state;
        Type = type;
    }

    public RepositoryOperation()
    {
    }

    protected virtual void JsonToResult(String value)
    {
    }

    protected virtual String ResultToJson()
    {
        return result;
    }
    
    public const String CREATED = "CREATED";
    public const String READY = "READY";
    public const String PENDING = "PENDING";
    public const String SUCCEDED = "SUCCEDED";
    public const String FAILED = "FAILED";
}

File: detecter/src/main/java/Model/RepositoryOperation.java
package Model;

public class RepositoryOperation {

  public static final String FAILED = "FAILED";
  public static final String SUCCEDED = "SUCCEDED";
  private String Repository;

  private String Type;

  private String State;

  private String Result;

  public RepositoryOperation(String repository, String type, String state, String result) {
    this.Repository = repository;
    this.Type = type;
    this.State = state;
    this.Result = result;
  }

  public String getRepository() {
    return Repository;
  }

  public void setRepository(String repository) {
    this.Repository = repository;
  }

  public String getType() {
    return Type;
  }

  public void setType(String type) {
    this.Type = type;
  }

  public String getState() {
    return State;
  }

  public void setState(String state) {
    this.State = state;
  }

  public String getResult() {
    return Result;
  }

  public void setResult(String result) {
    this.Result = result;
  }
}
